### DISCUSSION TITLE
Best way to spawn objects to move around?

### DISCUSSION THREAD

**Discussion by benthebear93** [google-deepmind/mujoco#2193] [Asking for Help]
### Intro

Hi!

I am a graduate student  at SDU Robotics, I use MuJoCo for my research on contact manipulation.

I am trying to spawn objects that i can move with robot manipulator. 
Usually i just add the object **body** with **freejoint**. 
I am also using mjcf from dm_control which asks for initial qpos
`arm_mjcf.keyframe.add("key", name="home", qpos=HOME_QPOS)`

### My setup

MuJoCo 3.2.0
Python
Linux 22.04

### My question

Although I could specify the joint positions for objects directly, things get quite messy when I want to spawn multiple objects or use an IK solver like [mink](https://github.com/kevinzakka/mink), because the IK solver will give you a solution for every joint. 

Everything makes sense logically but it doesn’t seem ideal to me. **Is there a better way to spawn objects that can move withoug adding extra qpos?**

### Confirmations

- [X] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

**Comment by yuvaltassa** [comment]
No. The answer is in the question "adding objects that can move" == "extra qpos"

**Reply by benthebear93** [reply]
Okay thank you for the answer :)
